sycl/ReleaseNotes.md

```diff
@@ -1104,7 +1104,7 @@ Release notes for commit range c9d50752..5d7e0925
   - The SYCL library doesn't guarantee stable API/ABI, so applications compiled
     with older version of the SYCL library may not work with new one.
     The workaround is to rebuild the application.
-    [ABI policy guide](doc/ABIPolicyGuide.md)
+    [ABI policy guide](doc/development-bkms/ABIPolicyGuide.md)
```

Comment:
Can we rename this to development-policies or development-guidelines ?
BKMs are something one may ignore, that's not what ABI doc is.

Reply:
I'm not opposed to changing the directory names.  Does everyone like the following:

* `sycl/doc/extensions` - Extension specifications
* `sycl/doc/design` - Design documents
* `sycl/doc/developer` - Developer information

Personally, I think it's better to keep end-user documentation directly in `sycl/doc` because it is more prominent.  I also think there are more likely to be external links to those documents, so there is some value to keeping them where they are.

Reply:
LGTM

Reply:
LGTM as well.

Reply:
I would like to hear from @alexbatashev to see if he agrees.  Updating all the links is kind of a pain, so I only want to do it once more.  :-)

Reply:
I renamed the directories in b00236cb4ca9286225e3626ff2854a4b16123d5c.  I also changed the titles in the Sphinx documentation in 429f2daa7341fb73367f4b1778ad14e7a35d57eb